bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/ui/view/visualizer/Visualizer.java

```diff
@@ -680,57 +681,57 @@ private void paintGroupBars(Graphics2D g2d) {
 	 * optionally as a closed shape corresponding to the area under the curve. These shapes
 	 * are cached.
 	 */
-	private Shape getElevationCurve(WorldCoords coords, long start, long end, boolean close, Map<WorldCoords, CachedShape> cache) {
+	private Shape getElevationCurve(Function<Long, WorldCoords> coords, long start, long end, boolean close, Map<WorldCoords, CachedShape> cache) {
 		synchronized (cacheLock) {
 
 			// Return the cached shape, if any.
-			CachedShape cs = cache.get(coords);
-			if (cs != null && cs.matches(start, end))
-				return cs.shape;
+//			CachedShape cs = cache.get(coords);
+//			if (cs != null && cs.matches(start, end))
+//				return cs.shape;
```

Comment:
The cache is a `Map<WorldCoords, CachedShape>` so the value stored is the whole shape of the plot.  We would "just" need to change the map lookup key to a small collection of, say, 3 coords.  You're probably right though that we shouldn't bother if performance is good enough anyway.  The only way to know is to see how well it works for a full night's plan on moderately old hardware.  If we're not going to use the cache though, we should remove them.